gips/0082-substreams-on-the-network.md

```diff
@@ -0,0 +1,106 @@
+---
+GIP: "0082"
+Title: Integrate Substreams into The Graph Network
+Authors: Alexandre Bourget <alex@streamingfast.io>
+Created: 2024-10-27
+Updated: 2024-10-27
+Stage: Draft
+Discussions-To: *To be added later*
+Category: Protocol Logic, Protocol Interfaces
+Implementations: *To be added later*
+---
+
+*This document tries to express and summarize the discussions that happened in-person at the October 2024 R&D Retreat.*
+
+# Abstract
+
+This GIP proposes integrating Substreams, a streaming-first blockchain indexing technology, into The Graph Network. This integration will enable Substreams to be fully recognized on the network and eventually receive indexing rewards. This proposal, however, does not entail immediately enabling rewards, but consists of the first approval and direction setting.
+
+# Motivation
+
+Substreams provide significant performance improvements for indexing, particularly for large datasets and complex computations. Integrating Substreams into the network will allow developers to leverage these performance gains while maintaining the decentralization and security benefits of The Graph, as well as encouraging the market to use The Graph as its prime infrastructure provider.
+
+# Detailed Specification
+
+## Indexer Selection Algorithm (ISA)
+
+Given that there's no gateway involved with Substreams (it's direct point-to-point between consumer and provider), the existing Indexer Selection Algorithm (ISA) does not apply.  A round-robin selection of providers will be used in the front-end (https://thegraph.market) to ensure fair distribution of request loads, while still allowing consumers to choose a provider.
+
+## Payments
+
+Payments for Substreams work will be on-chain. The existing `collect` function in the staking contract will be used for payment processing.  It was deemed that no TAP was required to have Substreams collect payments and honor the promises of the protocol. However, as TAP brings augmented trust minimization properties, it would be incorporated in future work.
+
+## Economic Security and Disputes
+
+Economic security will be enforced through slashing and disputes. Indexers providing incorrect data can have their stake slashed.
```

Comment:
This ties to my previous comment, how is econ security enforced? Indexers need to just stake? Do they need to allocate in some form?